File: DSA/LeetCode/Hashmaps/409-Longest-Palindrome.java
import java.util.HashMap;

class Solution {
    public int longestPalindrome(String s) {

        HashMap<Character, Integer> map = new HashMap<>();
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
               int rep = map.getOrDefault(c, 0);
            map.put(c, rep + 1);

        }
        int count = 0;

        int OddsCount = 0;
        for (int k : map.values()) {

            if (k % 2 == 0) {
                count += k;
            } else {
                count += k - 1;
                OddsCount++;
            }
        }

        if (OddsCount > 0)
            count += 1;
        return count;

    }
}

File: DSA/LeetCode/Hashmaps/409-Longest-Palindrome.js
/**
 * @param {string} s
 * @return {number}
 */
var longestPalindrome = function (s) {

    const map = new Map()
    for (let i = 0; i < s.length; i++) {
        let rep = map.get(s[i]) ?? 0
        rep++;
        map.set(s[i], rep);
    }
    let count = 0

    let OddsCount = 0
    for (let k of map.values()) {
        if (k % 2 == 0) {
            count += k
        } else {
            count += k - 1
            OddsCount++
        }
    }

    if (OddsCount > 0)
        count += 1
    return count

};